### DISCUSSION TITLE
TypeError on JSON object

### DISCUSSION THREAD

**Discussion by seanlabs** [alpinejs/alpine#781] [4. General]
I'm using eager loading via Laravel to pull in models and corresponding models into a single JSON object. One of the relationships is a many to many, but the relationship does not always exist. For example, I have a many to many relationship for Post and Tag and for each Post there is not always a corresponding "Tag". So, when I try and filter on the JSON object, I receive an error "TypeError: undefined is not an object (evaluating 'post.tag.name'). It's because for that particular post there is not a corresponding tag... I can't seem to figure out how to make it ignore this in the filter function.

Is there a way to check for it being undefined and ignore inside of my filter function? Below is a snippet of my function that does the filtering of the posts to either all or current.

```

let postApp = function(posts) {
                return {
                    posts: posts,
                    filter: 'active',
			
get current() {
                        return this.posts.filter((tag) => (post.tag.name == 'New' || post.tag.name == 'Assigned'))
                    },

get filteredPosts() {
                        return {
                            all: this.posts,
                            current: this.current,
                        }[this.filter];
                    }

```

**Comment by SimoTod** [comment]
The filter callback doesn't look correct. 
You named the variable tag but you are trying to use post in your callback. 
Also, tag(s?) is probably an array from what you described so your filter callback should probably be something like
```javascript 
this.posts.filter(
    (post) => post.tags.some(
        (tag) => tag.name === 'New' || tag.name === 'Assigned' 
    )
)
```
